Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java

```diff
@@ -0,0 +1,263 @@
+            //Change this
+            logger.log(Level.WARNING, "No files found to process"); //NON-NLS
+            return iLeappFiles;
+        }
+
+        List<AbstractFile> iLeappFilesToProcess = new ArrayList<>();
+        for (AbstractFile iLeappFile : iLeappFiles) {
+            if ((iLeappFile.getName().toLowerCase().contains(".zip") || (iLeappFile.getName().toLowerCase().contains(".tar"))
+                    || iLeappFile.getName().toLowerCase().contains(".tgz"))) {
+                iLeappFilesToProcess.add(iLeappFile);
+            }
+        }
+
+        return iLeappFilesToProcess;
+    }
+
+    private ProcessBuilder buildiLeappCommand(Path moduleOutputPath, String sourceFilePath, String iLeappFileSystemType) {
+
+        ProcessBuilder processBuilder = buildProcessWithRunAsInvoker(
+                "\"" + iLeappExecutable + "\"", //NON-NLS
+                "-t", iLeappFileSystemType, //NON-NLS
+                "-i", sourceFilePath, //NON-NLS
+                "-o", moduleOutputPath.toString()
+        );
+        processBuilder.redirectError(moduleOutputPath.resolve("iLeapp_err.txt").toFile());  //NON-NLS
+        processBuilder.redirectOutput(moduleOutputPath.resolve("iLeapp_out.txt").toFile());  //NON-NLS
+        return processBuilder;
+    }
+
+    static private ProcessBuilder buildProcessWithRunAsInvoker(String... commandLine) {
+        ProcessBuilder processBuilder = new ProcessBuilder(commandLine);
+        /*
+         * Add an environment variable to force log2timeline/psort to run with
+         * the same permissions Autopsy uses.
+         */
+        processBuilder.environment().put("__COMPAT_LAYER", "RunAsInvoker"); //NON-NLS
+        return processBuilder;
+    }
+
+    private static File locateExecutable(String executableName) throws FileNotFoundException {
+        String executableToFindName = Paths.get(ILEAPP, executableName).toString();
+
+        File exeFile = InstalledFileLocator.getDefault().locate(executableToFindName, ILeappAnalyzerIngestModule.class.getPackage().getName(), false);
+        if (null == exeFile || exeFile.canExecute() == false) {
+            throw new FileNotFoundException(executableName + " executable not found.");
+        }
+        return exeFile;
+    }
+
+    /**
+     * Find the index.html file in the iLeapp output directory so it can be added to reports
+     */
+    private void addILeappReportToReports(Path iLeappOutputDir, Case currentCase) {
+        List<String> allIndexFiles = new ArrayList<>();
+
+        try (Stream<Path> walk = Files.walk(iLeappOutputDir)) {
+
+            allIndexFiles = walk.map(x -> x.toString())
+                    .filter(f -> f.toLowerCase().endsWith("index.html")).collect(Collectors.toList());
+
+            if (allIndexFiles.get(0).length() > 2) {
+                currentCase.addReport(allIndexFiles.get(0), MODULE_NAME, Bundle.ILeappAnalyzerIngestModule_report_name());
+            }
+
+        } catch (IOException | TskCoreException ex) {
```

Comment:
You also need to catch UncheckedIOException, a runtime wrapper for IOException. The IOException caught here, according to the Files.walk documentation, is just related to accessing the iLeappOutputDir. If there is a problem while iterating, then UncheckedIOException will be thrown. Checked exception support was never worked into Java 8 streams and lambdas. You can log the underlying IOException by calling ex.getCause() or something like that.

Reply:
Added exception